spmd/tensor/ops/tensor_ops.py

```diff
@@ -39,6 +39,7 @@ def no_shard_prop_rule(op_schema: OpSchema) -> OutputSharding:
     "aten.is_same_size.default",
     "aten.ones_like.default",
     "aten.new_empty_strided.default",
+    "aten._softmax.default",
```

Comment:
this is essentially a math op, not a tensor op, we should add this to the math_ops.py, i.e. just `@register_prop_rule` in math_ops.py for `aten._softmax.default` and `aten._softmax_backward_data`, note that in the rule you should explicitly check if the sharding dim is the same as softmax dim (if it is, we should error out for now)